engine/dlib/src/dlib/message.cpp

```diff
@@ -252,9 +245,13 @@ namespace dmMessage
 
     static MessageSocket* AcquireSocket(HSocket socket)
     {
-        DM_SPINLOCK_SCOPED_LOCK(g_MessageContext->m_Spinlock);
+        DM_SPINLOCK_SCOPED_LOCK(g_MessageContext.m_Spinlock);
+        if (!g_MessageContextInitialized)
+        {
+            return 0;
+        }
```

Comment:
The problem was that the logging system might still be alive, and try to push messages, while the program is shutting down.